internal/datastore/crdb/caveat.go

```diff
@@ -9,17 +9,18 @@ import (
 )
 
 func (cr *crdbReader) ReadCaveatByName(ctx context.Context, name string) (*core.CaveatDefinition, datastore.Revision, error) {
-	return nil, datastore.NoRevision, fmt.Errorf("unimplemented caveat support in datastore")
+	return nil, datastore.NoRevision, fmt.Errorf("unimplemented caveat support in datastore for readcaveatbyname")
 }
 
 func (cr *crdbReader) ListCaveats(ctx context.Context, caveatNames ...string) ([]*core.CaveatDefinition, error) {
-	return nil, fmt.Errorf("unimplemented caveat support in datastore")
+	// Unimplemented, but returns empty to not break callers.
+	return nil, nil
```

Comment:
Why not returning a specific typed error that could be ignored by those callsites that need it?

Reply:
Because its a lot of work to check everywhere for little-to-no-gain; listing caveats when they are not supported doesn't really harm anything; its *writing* them or *referencing* them that will cause problems.